src/Simulator/IncrementalRedstoneSimulator/PressurePlateHandler.h

```diff
@@ -72,16 +78,134 @@ class cPressurePlateHandler : public cRedstoneHandler
 		UNUSED(a_PoweringData.PowerLevel);
 		// LOGD("Evaluating clicky the pressure plate (%d %d %d)", a_Position.x, a_Position.y, a_Position.z);
 
-		auto Power = GetPowerLevel(a_World, a_Position, a_BlockType, a_Meta);
-		auto PreviousPower = static_cast<cIncrementalRedstoneSimulator *>(a_World.GetRedstoneSimulator())->GetChunkData()->ExchangeUpdateOncePowerData(a_Position, PoweringData(a_BlockType, Power));
+		auto ChunkData = static_cast<cIncrementalRedstoneSimulator *>(a_World.GetRedstoneSimulator())->GetChunkData();
+
+		const auto PreviousPower = ChunkData->GetCachedPowerData(a_Position);
+		auto Power = GetPowerLevel(a_World, a_Position, a_BlockType, a_Meta);  // Get the current power of the platey
 
-		if (Power != PreviousPower.PowerLevel)
+		const auto PlateUpdates = GetAdjustedRelatives(a_Position, StaticAppend(GetRelativeLaterals(), cVector3iArray{ OffsetYM() }));
+		auto DelayInfo = ChunkData->GetMechanismDelayInfo(a_Position);
+
+		// Resting state?
+		if (DelayInfo == nullptr)
 		{
-			a_World.SetBlockMeta(a_Position, (Power == 0) ? E_META_PRESSURE_PLATE_RAISED : E_META_PRESSURE_PLATE_DEPRESSED);
-			return GetAdjustedRelatives(a_Position, StaticAppend(GetRelativeLaterals(), cVector3iArray{ OffsetYM() }));
+			if (Power == 0)
+			{
+				// Nothing happened, back to rest
+				return {};
+			}
+
+			// From rest, a player stepped on us
+			// Schedule a minimum 0.5 second delay before even thinking about releasing
+			ChunkData->m_MechanismDelays[a_Position] = std::make_pair(5, true);
+
+			// manage on-sound
+			AString soundToPlay = "";
```

Comment:
Agree with @tigerw. This should be its own `static` function:
```cpp
auto soundToPlay = GetClickOnSound(a_BlockType);
```
Same for the off sound.